### ISSUE TITLE
support deque in objgrep

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2323]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#2323|
|Type|enhancement|
|Created|2006-12-23 06:49:07Z|

This is an outgrowth of [#10745](https://github.com/twisted/twisted/issues/10745), but maybe it would be useful on its own.

Attachments:

* [test_reflect_2323.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f37/f37be93b674e3dcd988cba4a7cf66879468c3b35/6d2b8f320095f7952d72d6d6c1a4246117a1bd14.diff) (3480 bytes) - added by therve on 2007-01-12 11:29:00Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2323 2323
type__enhancement enhancement
reporter__glyph glyph
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1166856547000000 1166856547000000
changetime__1169919053000000 1169919053000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Committed to branch ```dqog-2323```.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to thomasvs|
|-|-|

At the very least, ```testDeque``` should be ```test_deque``` and have a docstring.  Are there any arguments against adding this though?  I personally can't think of any.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

I'd to use a bit of duck-typing here to test list-like objects and not rely on isinstance.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

You really need to do type checking here because that's the only way to be certain you are correctly accounting for references.

Unless you do something entirely different instead.

I'd like to see something based on gc.get_referents(), since that will work for pretty much any type.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve removed owner|
|-|-|

Added a few docstring, and change method names to conform new standard. The patch is against the branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [19486]) apply patch attached to ticket

Refs [#2323](https://github.com/twisted/twisted/issues/2323)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @therve|
|-|-|

Might be nice to update the types.ListType and types.TupleType on the changed line in reflect.py to just be list and tuple.

Changes all look good.  Make the above change if you want, and merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set status to `closed`|
|-|-|

(In [19490]) Merge dqog-2323

Author: therve
Reviewer: exarkun
Fixes [#2323](https://github.com/twisted/twisted/issues/2323)

Add support for deque in reflect.objgrep.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
